lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -910,15 +910,44 @@ RequirementRequest::evaluate(Evaluator &evaluator,
 }
 
 llvm::Expected<Type>
-swift::StructuralTypeRequest::evaluate(Evaluator &evaluator,
-                                       TypeAliasDecl *D) const {
-  TypeResolutionOptions options(TypeResolverContext::TypeAliasDecl);
-  if (!D->getDeclContext()->isCascadingContextForLookup(
-        /*functionsAreNonCascading*/true)) {
+StructuralTypeRequest::evaluate(Evaluator &evaluator,
+                                TypeAliasDecl *typeAlias) const {
+  // Fast path: If the interface type is already resolved, there's no need
+  // to compute the structural type.  This also prevents us from writing
+  // ErrorTypes into otherwise valid ASTs with generated typealiases.
+  if (typeAlias->hasInterfaceType()) {
+    return typeAlias->getInterfaceType()->getMetatypeInstanceType();
```

Comment:
This should use getDeclaredInterfaceType()

Reply:
Good catch